server/service/osquery.go

```diff
@@ -1703,35 +1703,68 @@ func getMostRecentResults(results []*fleet.ScheduledQueryResult) []*fleet.Schedu
 	return filteredResults
 }
 
-// Query names recieved from osqueryd are prefixed by teamID so we need
-// to pull them out to match the query name and team ID in the database
+// findPackDelimiterString attempts to find the `pack_delimiter` string in the scheduled
+// query name reported by osquery (note that `pack_delimiter` can contain multiple characters).
+//
+// The expected format for s is "pack<pack_delimiter>{Global|team-<team_id>}<pack_delimiter><query_name>"
+//
+// Returns "" if it failed to parse the pack_delimiter.
+func findPackDelimiterString(scheduledQueryName string) string {
+	// Go's regexp doesn't support backreferences so we have to perform some manual work.
+	scheduledQueryName = scheduledQueryName[4:] // always starts with "pack"
+	for l := 1; l < len(scheduledQueryName); l++ {
+		sep := scheduledQueryName[:l]
+		rest := scheduledQueryName[l:]
+		pattern := fmt.Sprintf(`^(?:(Global)|(team-\d+))%s.+`, regexp.QuoteMeta(sep))
```

Comment:
What if the delimiter has `Global` in its name?

Reply:
I believe this covers that scenario, see the added tests.

Reply:
Specifically:
`packGlobalGlobalGlobalGlobalGlobalGlobal`
Is delimiter `Global` or `GlobalGlobal`?
Granted, this is a corner case.

Reply:
Ah good catch, it breaks if separator is `GlobalGlobal`. Yeah moving forward we should prevent users from setting pack_delimiters like that that break the system.